examples/erc20/src/lib.rs

```diff
@@ -0,0 +1,298 @@
+        Ok(self.total_supply)
+    }
+
+    /// Add admin
+    pub fn add_admin(&mut self, ctx: &Context, admin: Address) -> Result<()> {
+        if !self.admins.contains(&ctx.sender()) {
+            return Err(Error::AdminPrivilegesRequired);
+        }
+        self.admins.insert(admin);
+        Ok(())
+    }
+}
+
+// Helper methods
+
+/// Balance of
+fn _balance(accounts: &mut HashMap<Address, u64>, addr: Address) -> u64 {
+    match accounts.entry(addr) {
+        Entry::Vacant(_ve) => return 0u64,
+        Entry::Occupied(oe) => return *oe.get(),
+    }
+}
+
+/// transfer method
+fn _transfer(
+    accounts: &mut HashMap<Address, u64>,
+    from: Address,
+    to: Address,
+    amount: u64,
+) -> bool {
+    let from_balance = _balance(accounts, from);
+    let to_balance = _balance(accounts, to);
+
+    // check for sufficient balance
+    if from_balance < amount {
+        return false;
+    }
+    accounts.insert(from, from_balance - amount);
+    accounts.insert(to, to_balance + amount);
+
+    Event::emit(&Transfer {
+        from,
+        to,
+        amount,
+    });
+
+    true
+}
+
+impl ERC20Token {
+    /// transfer
+    pub fn transfer(&mut self, ctx: &Context, to: Address, amount: u64) -> Result<Transfer> {
+        let from = ctx.sender();
+        if from == to || amount == 0u64 {
+            // no-op
+            return Ok(Transfer::default());
+        }
+        if _transfer(&mut self.accounts, ctx.sender(), to, amount) {
+            return Ok(Transfer {
+                from,
+                to,
+                amount
+            });
+        }
+        Err(Error::InsufficientFunds { address: from })
+    }
+
+    /// allowance
+    pub fn approve(&mut self, ctx: &Context, spender: Address, amount: u64) -> Result<Approval> {
+        let allowances = match self.allowed.entry(ctx.sender()) {
+            Entry::Vacant(ve) => ve.insert(HashMap::new()),
+            Entry::Occupied(oe) => oe.into_mut(),
+        };
+        allowances.insert(spender, amount);
+
+        let approval = Approval {
+            sender: ctx.sender(),
+            spender,
+            amount,
+        };
+
+        Event::emit(&approval);
+
+        Ok(approval.clone())
+    }
+
+    /// read allowance
+    pub fn allowance(&mut self, ctx: &Context, spender: Address) -> Result<u64> {
+        if !self.allowed.contains_key(&ctx.sender()) {
+            return Ok(0u64);
+        }
+        let allowances = self.allowed.get_mut(&ctx.sender()).unwrap();
+        if !allowances.contains_key(&spender) {
+            return Ok(0u64);
+        }
+        Ok(*allowances.get(&spender).unwrap())
+    }
+
+    /// transfer from a given account up to the given allowance
+    pub fn transfer_from(
+        &mut self,
+        _ctx: &Context,
+        from: Address,
+        spender: Address,
+        amount: u64,
+    ) -> Result<Transfer> {
+        let allowances = self.allowed.get_mut(&from).unwrap();
```

Comment:
I don't think that this has to be `mut`. Especially since you deref it a few lines below.

Reply:
I am de-reffing the result of the `get(&spender)` post `unwrap` which is a `&u64`. Compiler complains if I remove the `mut`.